sky/backends/backend_utils.py

```diff
@@ -699,6 +699,8 @@ def write_cluster_config(to_provision: 'resources.Resources',
         auth_config = onprem_utils.get_local_auth_config(cluster_name)
     region_name = resources_vars.get('region')
 
+    sky_wheel_hash = os.path.basename(local_wheel_path)
```

Comment:
This is leaking knowledge across modules/funcs, can be brittle. Can we add to `RetryingVmProvisioner` a `self._local_wheel_hash`, returned from `wheel_utils`?